### PR TITLE
Fix #515. Don't accept unescaped control characters

### LINKED ISSUE #515: Don't accept unescaped control characters, as per RFC 7159
The changelog [mentions](https://github.com/bos/aeson/blob/master/changelog.md#1100) this isn't done yet. Is there any objection to doing so? I think accepting a document as JSON when it actually isn't is a fairly big deal.

### PR DESCRIPTION
**Pull request by phadej**
I consider this as a breaking change.

### CODE REVIEW

File: aeson.cabal [hunk 1]
```diff
@@ -175,7 +175,7 @@ library
     hs-source-dirs: ffi
     other-modules: Data.Aeson.Parser.UnescapeFFI
 
-test-suite tests
+test-suite aeson-tests
```
Comment (bergmark) [inline aeson.cabal:178]: I think this name is mentioned in CONTRIBUTING.md

Comment (phadej) [reply aeson.cabal:178]: Updated.

**Comment by phadej** [comment]
CI green

**Comment by bergmark** [comment]
Nice!

**Pull request merged**